configure

```diff
@@ -276,41 +278,39 @@ elif (args.eigen != ""):
     f.write("EIGEN = " + args.eigen + "\n")
 
 #
-# Compiler
+# Compiler and MPI library
 #
 if not args.comp in ['g++','icc','clang++']:
     raise Exception(color.red + "ERROR" + color.reset + ": Compiler must be g++, clang++, or icc")
-if (args.comp == "gcc"): f.write("COMP = GCC\n")
-elif (args.comp == "clang"): f.write("COMP = CLANG\n")
-elif (args.comp == "intel"): f.write("COMP = INTEL\n")
-message("Comp-Type", str(args.comp))
-shellmessage("Comp-Version",args.comp+" --version")
-shellmessage("Comp-MPI","mpicxx -cxx="+args.comp+" -show")
-f.write("CC = mpicxx -cxx="+args.comp+"\n")
-if args.comp == 'g++':     
-    f.write("MPI_LIB = -lgfortran -lmpich\n")
+
+if not args.alternate_mpi and not shutil.which("mpichversion"):
+    raise Exception(color.red + "ERROR" + color.reset + ": An MPICH installation was not found, to compile with another MPI library please use " + code("--alternate-mpi"))
+
+shellmessage("Compiler (" + args.comp + ")", args.comp + " --version")
+shellmessage("MPI Library","mpirun --version")
```

Comment:
Leave as MPICH please. You can add other behavior to be triggered by `args.alternate_mpi` if you like. But default should be MPICH.

Also why did you swith to `mpirun --version`? This is much less informative for compilation purposes. Please stick with the original output.

Reply:
Default _is_ MPICH/MVAPICH.

Because that's supported by all MPI libraries. What information are you missing now?